### ISSUE TITLE
My Conversations > Open Conversations > Separate Supv and Employee input fields

### ISSUE THREAD

**Issue by Travis-A-Clark** [bcgov/performance#736]

**Comment by Travis-A-Clark** [comment]
Just a note for myself: ask Devs about auto expanding text boxes as content is entered rather than small input area and scroll as currently displayed.

**Comment by Travis-A-Clark** [comment]
@jchow11 Just two questions:

1) Are we still okay to put in all the text box descriptions as info toggles rather than displaying all the text by default? This is what we primarily use right now. I am wary of making the template look too overwhelming with all the text.

2) Please remove the "disagree" option from the supervisor side of the sign-off. It's just an option for employees.

**Comment by jchow11** [comment]
@Travis-A-Clark My suggestion is to use toggletips as sparingly as possible– often information within them is missed. As well, users cannot view the toggletip content and type in comments at the same time. While it can be overwhelming to read a lot of text, it can also be helpful for users to be able to directly reference the helper text as they are entering in comments– this way they know they are correctly answering each section.

That being said, we can definitely change the helper text to "i" toggletips if you feel our users will be overwhelmed! Was there specific feedback we received in initial testing of the conversations section that sways you one way or the other?

**Comment by jchow11** [comment]
@Travis-A-Clark I am seeing now that in the rest of the app, you are able to keep the toggle tip open and type in a comment at the same time. My suggestion is still to keep everything visible as toggletip information is easy to miss, but if we want to go the toggletip route, I would suggest that at least some of the current text remains visible as helper text. My reasoning for this is that the input names are not super clear without extra text – "Appreciation", "Coaching", "Evaluation". The user cannot glean what to input in these areas without helper text, so it should be clearly presented to the user. Supplemental info could be included in a toggletip, but the user should be able to understand what to enter without having to click on it.

**Comment by Travis-A-Clark** [comment]
@jchow11 Yes, the hybrid approach in your last comment is exactly what I had in mind. :-) In the word doc that I attached, the intent was that the plain text descriptions would be visible and the bolded descriptions would be in a toggle tip. If that balance works for you, I'm happy to move forward with this approach for the other templates. Thanks!

**Comment by jchow11** [comment]
@Travis-A-Clark I think it's a good solution! I'll move forward with that and update the wireframes. Thanks!

**Comment by jchow11** [comment]
@Travis-A-Clark would you like me to create mockups for all of the templates, or just one as an example?

**Comment by Travis-A-Clark** [comment]
@jchow11 I think it would be good to at least do the Career Development and Performance Improvement mock ups because there is some substantial change in the data capture section for both.

**Comment by jchow11** [comment]
@Travis-A-Clark ticket is updated.

**Comment by jizhaogit** [comment]
Hi,  we have some topics that have session named "Information to Capture",  which are 
- Performance Check-In
- Performance Improvement
- Career Development

i cannot find this session from the new design.  please verify if we need to worry about it.  Thanks

**Comment by Travis-A-Clark** [comment]
@jizhaogit No need to worry about that section. It has been replaced in the new design with the Employee Comments and Supervisor Comments section. As long as we've got all the headings and text boxes from the Word doc attached to this ticket in those two areas, we are good to go. Thanks!

**Comment by jizhaogit** [comment]
Thanks Travis,  i have another question.  why we have both supervisor employee-id sign-off and employee employee-id sign-off input filed show in the conversation modal? should that be shown ONLY based on the accessor's role?  thanks

**Comment by jizhaogit** [comment]
okay will go this way

**Comment by steffipravasi** [comment]
@Travis-A-Clark Are we keeping the text in the messages/errors the same as current? Also, should we change the title of the message as right now it shows the site.
![Screen Shot 2022-08-25 at 9.48.54 AM.png](https://images.zenhubusercontent.com/62e1b008ce7d6652fe102828/95e170b6-7aad-4f60-9185-fb118de4e712)

**Comment by jchow11** [comment]
@Travis-A-Clark @jizhaogit updated the ticket with all of the conversation template designs. Travis– the word doc only specified a single text area for "Comments and action items", although in the other templates these are 2 separate inputs. I've kept them as 2 separate inputs as per the other templates, but let me know if you'd rather them be combined as per the word doc. Thanks!

**Comment by Travis-A-Clark** [comment]
@jchow11 Looks good! Thanks for keeping those areas separate.

@jizhaogit Please go ahead and finish off the dev work with the new designs. Thanks!

**Comment by jchow11** [comment]
@jizhaogit Alison flagged a spelling mistake on the Goal Setting Template, could you address this while you're doing this work? Just a small typo :)

![MicrosoftTeams-image.png](https://images.zenhubusercontent.com/61e98ff375d0505e601dd00e/b8a93ef4-011b-4376-aa92-0dc6111c451c)

**Comment by jizhaogit** [comment]
sure will do

**Comment by steffipravasi** [comment]
Test status: Fail
@jizhaogit Please refer to the [Test cases document ](https://bcgov.sharepoint.com/:w:/r/teams/02915/_layouts/15/Doc.aspx?sourcedoc=%7BF61FC20A-96A6-487A-BD67-30267E628D25%7D&file=736.docx&action=edit&mobileredirect=true&wdPreviousSession=b7beb568-0dc9-4510-875b-ce3f8b1dfa57&wdOrigin=TEAMS-ELECTRON.teamsSdk.undefined&cid=bb8de9c4-0b55-4ba3-a0ea-0e4cea6fffec) for the fixes. You can find the defect numbers to be fixed at the very top the document in the table. You can refer to the test cases below them to view the detailed defect information.

**Comment by steffipravasi** [comment]
Test status: Pass
Env: Test
The defects in the documents has been fixed. Ticket ready for PO review.

**Comment by jizhaogit** [comment]
fixing for user name sign (test case #20) is addressed.

**Comment by steffipravasi** [comment]
Minor issue needs to be fixed. Assigning back to @jizhaogit

**Comment by jizhaogit** [comment]
please try again, the issue should be addressed.

**Comment by Travis-A-Clark** [comment]
@steffipravasi I found one issue. Travis Clark (supervisor) sets up a conversation with Zehra Pirani LeRoy (employee). Both fill out info. Zehra checks the "disagree" box and then signs off. Travis goes to sign off, and Travis has the ability to select/unselect the "disagree" box that Zehra checked (see image below). This box should only ever be clickable by the employee and should be locked in once the employee signs. Can you see if there's anythin else similar and then work with Ji to correct? Thanks!

![image.png](https://images.zenhubusercontent.com/5ffe1f472e351dd9b643025b/2fdaabcb-f69f-4d0c-a0a2-1752690b684c)

**Comment by jizhaogit** [comment]
the disagree checkbox issue is addressed

**Comment by steffipravasi** [comment]
@jizhaogit I see that the disagree checkbox is fixed for the employee not but there is one more issue here. The disagree checkbox is disabled for the  employee if the supervisor signs the conversation before the employee. It only gets enabled if the supervisor has not signed or un-signs the convo. 
![Screen Shot 2022-09-23 at 11.07.53 AM.png](https://images.zenhubusercontent.com/62e1b008ce7d6652fe102828/000a37af-8628-4bf7-bcd8-f5913ee2ada2)

**Comment by jizhaogit** [comment]
it is addressed. thanks!

**Comment by steffipravasi** [comment]
Issue has been resolved. Ticket ready for PO review.
